src/interfaces/ICycleModule.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "./ICycleManager.sol";
+
+/// @title ICycleModule
+/// @notice Interface for pluggable cycle management strategies
+/// @dev Implementations of this interface provide different cycle timing mechanisms
+interface ICycleModule {
+    /// @notice Emitted when the cycle length is updated
+    /// @param oldLength The previous cycle length
+    /// @param newLength The new cycle length
+    event CycleLengthUpdated(uint256 oldLength, uint256 newLength);
+
+    /// @notice Emitted when a new cycle is started
+    /// @param cycleNumber The number of the new cycle
+    /// @param startBlock The block when the cycle started
+    event NewCycleStarted(uint256 indexed cycleNumber, uint256 startBlock);
+
+    /// @notice Gets the current cycle number
+    /// @return The current cycle number
+    function getCurrentCycle() external view returns (uint256);
+
+    /// @notice Checks if distribution is ready based on the cycle timing
+    /// @return Whether the cycle timing allows for distribution
+    function isDistributionReady() external view returns (bool);
```

Comment:
if this is implemented here then it seems it should not be in the manager , we may want to consider a refactor to not have a manager but instead of an abstract module and conslidate all this functionality